tests/unit/authenticators/oauth2-password-grant-test.js

```diff
@@ -228,6 +238,38 @@ describe('OAuth2PasswordGrantAuthenticator', () => {
         });
       });
     });
+
+    describe('when the authentication request fails without a valid response', function() {
+      beforeEach(function() {
+        server.post('/token', () => [500, { 'Content-Type': 'text/plain', 'X-Custom-Context': 'foobar' }, 'The server has failed completely.']);
+      });
+
+      it('rejects with the response body', function(done) {
```

Comment:
No need to use `done` here - you can `return authenticator.authenticate(…` instead.

Reply:
I copied this from another example in this file, but I'm pretty sure this is necessary. The assertions are in the `catch` on the promise, so if the promise doesn't reject, the test will pass without any assertions being made.

Reply:
ah that's a very good point :+1: